shell/platform/android/io/flutter/embedding/engine/renderer/FlutterRenderer.java

```diff
@@ -589,11 +585,9 @@ private void maybeWaitOnFence(Image image) {
         waitOnFence(image);
         return;
       }
-      if (!ignoringFence) {
```

Comment:
What happened here?

The instance starts with "ignoringFence = false" and this check is to ensure we only log the message once.

Reply:
Line 584 says:

```java
if (ignoringFence) {
  return;
}
```

So in what case will `if (ignoringFence)` hold on line 592? Maybe I'm missing something subtle about multi-threaded access?

Reply:
Doh! I missed that if at the top. Sorry for the noise.